### PR TITLE
Add in guild_create event task

### PR DESCRIPTION
**Pull request by sahil-kale**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: src/listeners/guild_create.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import colors from '../utils/colorUtil';
+import { Guild } from 'discord.js';
+import { surroundStringWithBackTick } from '../utils/discordUtil';
+import {getSettings } from '../database/models/SettingsModel';
+import settingsModel from "../database/models/SettingsModel";
+
+module.exports = class GuildCreateListener extends Listener {
+  constructor(context: PieceContext) {
+    super(context, {
+      once: true,
```
Comment (zaydtahir) [inline src/listeners/guild_create.ts:12]: This should not be once, once means it will only ever fire once

Comment (zaydtahir) [reply src/listeners/guild_create.ts:12]: Just remove this line

File: src/listeners/guild_create.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      event: Events.GuildCreate,
+    });
+  }
+
+  public async run(guild_object: Guild) {
+    let channel = guild_object.systemChannel
```
Comment (zaydtahir) [inline src/listeners/guild_create.ts:18]: This is the channel that those discord made info messages get sent right?

have you checked that this works even if a server has that setting off?

File: src/listeners/guild_create.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    await settingsModel.deleteOne({_id: guild_object.id}) //This will delete existing settings data
+    const settingsData = await getSettings(guild_object.id as string);
+    const embed = new BediEmbed()
+      .setTitle('BediBot has entered the server!')
+      .setColor(colors.PRIMARY)
```
Comment (zaydtahir) [inline src/listeners/guild_create.ts:23]: Unnecessary, BediEmbed by default will set the color to this.

File: src/listeners/guild_create.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+module.exports = class GuildCreateListener extends Listener {
+  constructor(context: PieceContext) {
+    super(context, {
+      once: true,
+      event: Events.GuildCreate,
+    });
+  }
+
+  public async run(guild_object: Guild) {
```
Comment (zaydtahir) [inline src/listeners/guild_create.ts:17]: Can we use camelCase please?

**Review by zaydtahir — CHANGES_REQUESTED** [review]

File: src/listeners/guild_create.ts (hunk 4 above)
Comment (sahil-kale) [reply src/listeners/guild_create.ts:17]: Pascal Case ftw

**Review by zaydtahir — APPROVED** [review]

**Pull request merged**